model/fn-execution/build.gradle

```diff
@@ -17,17 +17,14 @@
  */
 
 apply from: project(":").file("build_rules.gradle")
-applyJavaNature(enableFindbugs: false, enableErrorProne: false)
-applyGrpcNature()
+applyPortabilityNature()
 
 description = "Apache Beam :: Model :: Fn Execution"
 ext.summary = "Portable definitions for execution user-defined functions."
 
 dependencies {
-  compile library.java.guava
-  shadow project(path: ":beam-model-pipeline", configuration: "shadow")
-  shadow library.java.protobuf_java
-  shadow library.java.grpc_core
-  shadow library.java.grpc_protobuf
-  shadow library.java.grpc_stub
+  // We purposely depend on the unshaded classes for compilation and
```

Comment:
Can you explain how this works ? Won't compiled classes refer to unshaded classes ?

Reply:
I couldn't figure out a way to have the generated source use a vendored version of grpc/protobuf. So the generated code uses imports like `com.google.protobuf.Message` and expects classes from `beam-model-pipeline` to extend non-vendored gRPC/protobuf classes.

The flow is:
1) generate code using non-vendored imports
2) compile code using non-vendored dependencies
3) produce shaded classes using the same shading rules as `beam-model-pipeline`

This works because `beam-model-pipeline`, `beam-model-job-management`, and `beam-model-fn-execution` all use the same relocation rules so when the jar is loaded by java, java will see that they all extend the same vendored code locations.